### PR TITLE
INTMDB-802: Add Data Federation Query Limit

### PR DESCRIPTION
**Pull request by maastha**
## Description

Add Data Federation Query Limit to go-client.

Link to any related issue(s): [INTMDB-802](https://jira.mongodb.org/browse/INTMDB-802)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/data_federation.go [hunk 1]
```diff
@@ -33,6 +34,10 @@ type DataFederationService interface {
 	Create(context.Context, string, *DataFederationInstance) (*DataFederationInstance, *Response, error)
 	Update(context.Context, string, string, *DataFederationInstance, *DataFederationUpdateOptions) (*DataFederationInstance, *Response, error)
 	Delete(context.Context, string, string) (*Response, error)
+	ListQueryLimits(context.Context, string, string) ([]*DataFederationQueryLimit, *Response, error)
+	GetQueryLimit(context.Context, string, string, string) (*DataFederationQueryLimit, *Response, error)
+	ConfigureQueryLimit(context.Context, string, string, string, *DataFederationQueryLimit) (*DataFederationQueryLimit, *Response, error)
```
Comment (andreaangiolillo) [inline mongodbatlas/data_federation.go:39]: Generally, we use Update for `Patch`
```suggestion
	UpdateQueryLimit(context.Context, string, string, string, *DataFederationQueryLimit) (*DataFederationQueryLimit, *Response, error)
```

File: mongodbatlas/data_federation.go [hunk 2]
```diff
@@ -271,3 +288,112 @@ func (s *DataFederationServiceOp) Delete(ctx context.Context, groupID, name stri
 
 	return resp, err
 }
+
+// ConfigureQueryLimit Creates or updates one query limit for one federated database instance.
```
Comment (andreaangiolillo) [inline mongodbatlas/data_federation.go:292]:
```suggestion
// ConfigureQueryLimit updates one query limit for one federated database instance.
```

File: mongodbatlas/data_federation.go [hunk 3]
```diff
@@ -271,3 +288,112 @@ func (s *DataFederationServiceOp) Delete(ctx context.Context, groupID, name stri
...
+	}
+	if name == "" {
+		return nil, nil, NewArgError("name", "must be set")
+	}
+	if limitName == "" {
+		return nil, nil, NewArgError("limitName", "must be set")
+	}
+	if queryLimit == nil {
+		return nil, nil, NewArgError("queryLimit", "must be set")
+	}
```
Comment (andreaangiolillo) [inline mongodbatlas/data_federation.go:307]: I don't see `queryLimit` in https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Data-Federation/operation/createOneDataFederationQueryLimit. Is it missing in the doc?

**Review by andreaangiolillo — COMMENTED** [review]
left a few minor comments. Great work! 💯

File: mongodbatlas/data_federation.go (hunk 2 above)
Comment (maastha) [reply mongodbatlas/data_federation.go:292]: The same PATCH API is used to create and update this entity.

File: mongodbatlas/data_federation.go (hunk 1 above)
Comment (maastha) [reply mongodbatlas/data_federation.go:39]: @andreaangiolillo The same PATCH API is used to create and update this entity. Should we still change this to Update? Could that be misleading?

File: mongodbatlas/data_federation.go (hunk 3 above)
Comment (maastha) [reply mongodbatlas/data_federation.go:307]: This is basically for the Request Body Schema for this API, which is the same data structure as this entity in the Open API spec. Let me know if this should be handled differently.

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**